virttest/qemu_vm.py

```diff
@@ -3902,6 +3902,13 @@ def migrate(self, timeout=virt_vm.BaseVM.MIGRATE_TIMEOUT, protocol="tcp",
                         msg = ("Migrate capability '%s' should be '%s', "
                                "but actual result is '%s'" % (key, state, s))
                         raise exceptions.TestError(msg)
+                    clone.monitor.set_migrate_capability(state, key)
+                    s = clone.monitor.get_migrate_capability(key)
+                    if s != state:
+                        msg = ("Migrate capability '%s' should be '%s', "
+                               "but actual result is '%s' on destination guest"
+                               % (key, state, s))
+                        raise exceptions.TestError(msg)
```

Comment:
Just double-checking, is it possible to set "any" migraty capability on dst as well as on source and won't it affect the next migration? (in case we do multiple migrations with different params)

Reply:
Hmm, this can actually break something. On the other hand I'm not aware of any test that would do such thing without setting all the capabilities before migration so let's go with this and eventually fix the places where "default" capabilities without specifying them after migration with extra capabilities is used.

Btw that looks like a nice test, something like random kinds of migrations with various caps and params.